scripts/missions/amk/helpers.lua

```diff
@@ -52,6 +52,45 @@ xi.amk.helpers.helmTrade = function(player, helmType, broke)
     end
 end
 
+-- AMK 6 (index 5) - An Errand! The Professor's Price
+-- Cardian orb KI logic: All or nothing drop of the orb is handled by
+-- a local var that is set once by the first player is called by onMobDeath
+-- The rest of the players in alliance get the same outcome as the first
+xi.amk.helpers.cardianOrbDrop = function(mob, player, orb)
+    if
+        player:getCurrentMission(xi.mission.log_id.AMK) < xi.mission.id.amk.AN_ERRAND_THE_PROFESSORS_PRICE or
+        player == nil or
+        orb == nil
```

Comment:
This new change won't work, because you're checking the player's mission before you're checking if player is really there.
Lua properly has short-circuit logic, so you can just move the check for player or orb being nil before the mission check and you'll be fine.

https://www.tutorialspoint.com/execute_lua_online.php :
```lua
local player = nil

if player == nil or player:thisFunctionDoesntExist() == true then
    print("Short circuit success")
end
```

Prints correctly.

```lua
    if
        player == nil or
        player:getCurrentMission(xi.mission.log_id.AMK) < xi.mission.id.amk.AN_ERRAND_THE_PROFESSORS_PRICE or
```

You don't need to check whether or not orb is nil, because you're the one making the call and you know you've populated it

Reply:
D'oh!  Understood, thanks.  Will fix soon.